lib/torrent.js

```diff
@@ -410,65 +410,68 @@ export default class Torrent extends EventEmitter {
 
     const urls = Array.isArray(this.xs) ? this.xs : [this.xs]
 
+    self._xsRequestsController = new AbortController()
+
+    const signal = self._xsRequestsController.signal
+
     const tasks = urls.map(url => cb => {
       getMetadataFromURL(url, cb)
     })
     parallel(tasks)
 
-    function getMetadataFromURL (url, cb) {
+    async function getMetadataFromURL (url, cb) {
       if (url.indexOf('http://') !== 0 && url.indexOf('https://') !== 0) {
         self.emit('warning', new Error(`skipping non-http xs param: ${url}`))
         return cb(null)
       }
 
       const opts = {
-        url,
         method: 'GET',
         headers: {
           'user-agent': USER_AGENT
-        }
+        },
+        signal
       }
-      let req
+      let res
       try {
-        req = get.concat(opts, onResponse)
+        res = await fetch(url, opts)
       } catch (err) {
-        self.emit('warning', new Error(`skipping invalid url xs param: ${url}`))
+        self.emit('warning', new Error(`http error from xs param: ${url}`))
         return cb(null)
       }
 
-      self._xsRequests.push(req)
+      if (self.destroyed) return cb(null)
+      if (self.metadata) return cb(null)
 
-      function onResponse (err, res, torrent) {
-        if (self.destroyed) return cb(null)
-        if (self.metadata) return cb(null)
-
-        if (err) {
-          self.emit('warning', new Error(`http error from xs param: ${url}`))
-          return cb(null)
-        }
-        if (res.statusCode !== 200) {
-          self.emit('warning', new Error(`non-200 status code ${res.statusCode} from xs param: ${url}`))
-          return cb(null)
-        }
-
-        let parsedTorrent
-        try {
-          parsedTorrent = parseTorrent(torrent)
-        } catch (err) {}
+      if (res.status !== 200) {
+        self.emit('warning', new Error(`non-200 status code ${res.status} from xs param: ${url}`))
+        return cb(null)
+      }
+      let torrent
+      try {
+        // TODO: switch to uint8array once parseTorrent is updated
+        torrent = Buffer.from(await res.arrayBuffer())
+      } catch (e) {
+        self.emit('warning', e)
```

Comment:
not really? this will pretty much never fail, and the error would be as descriptive as it gets, as I can't really think of any issue which would cause this to fail to inform the user